contracts/ERC5018.sol

```diff
@@ -31,52 +26,53 @@ contract ERC5018 is LargeStorageManager, BlobStorageManager, IERC5018, ISemver {
         return storageModes[keccak256(name)];
     }
 
-    function _setStorageMode(bytes memory name, StorageMode mode) internal {
-        storageModes[keccak256(name)] = mode;
-    }
-
     // Large storage methods
     function write(bytes memory name, bytes calldata data) public onlyOwner payable virtual override {
         // TODO: support multiple chunks
         return writeChunk(name, 0, data);
     }
 
     function read(bytes memory name) public view virtual override returns (bytes memory, bool) {
-        StorageMode mode = getStorageMode(name);
+        bytes32 key = keccak256(name);
```

Comment:
Suggest to abstract the mode comparison logic since it repeats in many functions.